File: src/my_cpp_pkg/src/publish_node_customInterface.cpp
#include "rclcpp/rclcpp.hpp"
#include "my_robot_interfaces/msg/hardware_status.hpp"

class HardwareStatusPublisherNode : public rclcpp::Node
{
public:
    HardwareStatusPublisherNode() : Node("hardware_status_publisher")
    {
        RCLCPP_INFO(this->get_logger(), "Hardware status publisher has been started.");
        publisher_ = this->create_publisher<my_robot_interfaces::msg::HardwareStatus>("hardware_status", 10);
        timer_ = this->create_wall_timer(
            std::chrono::seconds(1),
            std::bind(&HardwareStatusPublisherNode::publishStatus, this));
    }

private:
    void publishStatus()
    {
        auto msg = my_robot_interfaces::msg::HardwareStatus();
        msg.version = 1;
        msg.temperature = 34.5;
        msg.are_motors_ready = true;
        msg.debug_message = "nothing to report";
        publisher_->publish(msg);
    }

    rclcpp::Publisher<my_robot_interfaces::msg::HardwareStatus>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<HardwareStatusPublisherNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/publish_node_customInterface.py
#! /usr/bin/env python3

import rclpy
from rclpy.node import Node
from my_robot_interfaces.msg import HardwareStatus



class NumberPublisherNode(Node):
    def __init__(self):
        super().__init__("temperature_publisher_customizeInterface")
        # self.number_ = 2
        self.number_publisher_ = self.create_publisher(HardwareStatus, "temperature", 10)
        self.timer_ = self.create_timer(1.0, self.publish_number)
        self.get_logger().info("Number publisher using customize Interfaces has been started.")

    def publish_number(self):
        msg = HardwareStatus()
        msg.temperature = 34.5
        self.number_publisher_.publish(msg)


def main(args=None):
    rclpy.init(args=args)
    node = NumberPublisherNode()
    node.publish_number()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()


if __name__ == "__main__":
    main()
